packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -293,26 +363,42 @@ function isFunctionParam(
   return false;
 }
 
-function isPromiseCatchCallWithHandler(
+function parseCatchCallWithHandler(
   expression: TSESTree.CallExpression,
-): boolean {
-  return (
+): { onRejected: TSESTree.CallExpressionArgument } | undefined {
```

Comment:
[Nitpick] Similar to https://github.com/typescript-eslint/typescript-eslint/pull/7092/files#r1262762950, we can simplify this type.

```suggestion
): TSESTree.CallExpressionArgument | undefined {
```

Reply:
Changed, and changed function names correspondingly